### PR TITLE
Fix minor implementation issues found in the JerryScript API.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by LaszloLango** [comment]
Please update the API reference too (based on the comment changes).

**Comment by LaszloLango** [comment]
LGTM

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -379,7 +379,8 @@ jerry_run_simple (const jerry_char_t *script_source,
 
 **Summary**
 
-Parse specified script to execute in Global scope.
+Parse script and construct an ecmascript function. The
```
Comment (akosthekiss) [inline docs/02.API-REFERENCE.md]: EcmaScript? (use CamelCase here as well?)

File: jerry-core/jerry.c [hunk 2]
```diff
@@ -1614,16 +1649,16 @@ jerry_get_object_native_handle (const jerry_value_t obj_val, /**< object to get
...
  *
  * Note:
  *      If native handle was already set for the object, its value is updated.
  *
  * Note:
- *      If free callback is specified, it is set to be called upon specified JS-object is freed (by GC).
- *
- *      Otherwise, if NULL is specified for free callback pointer, free callback is not created
- *      and, if a free callback was added earlier for the object, it is removed.
+ *      If free callback is specified (non-NULL), it will be called
```
Comment (akosthekiss) [inline jerry-core/jerry.c]: "If a non-NULL free callback is specified"

File: jerry-core/jerry.c [hunk 3]
```diff
@@ -254,7 +263,8 @@ jerry_run_simple (const jerry_char_t *script_source, /**< script source */
 } /* jerry_run_simple */
 
 /**
- * Parse script for specified context
+ * Parse script and construct an ecmascript function. The lexical
```
Comment (akosthekiss) [inline jerry-core/jerry.c]: EcmaScript

File: jerry-core/jerry.c [hunk 4]
```diff
@@ -1010,7 +1034,7 @@ jerry_string_to_char_buffer (const jerry_value_t value, /**< input string value
 } /* jerry_string_to_char_buffer */
 
 /**
- * Checks whether the object or it's prototype has the given property.
+ * Checks whether the object or it's prototype objects has the given property.
```
Comment (akosthekiss) [inline jerry-core/jerry.c]: "object has" or "objects have"

**Comment by akosthekiss** [comment]
LGTM after the doc/comment fixes

**Pull request merged**